src/kernel_abi.h

```diff
@@ -1969,6 +1972,85 @@ struct X86Arch : public BaseArch<SupportedArch::x86, WordSize32Defs> {
   RR_VERIFY_TYPE_ARCH(SupportedArch::x86, struct ::stat64, struct stat64);
 };
 
+// Archs that inherit Linux's "generic" data structures
+template <SupportedArch arch_, typename wordsize>
+struct GenericArch : public BaseArch<arch_, wordsize> {
+  // All architectures using the generic syscall table only ever use 32-bit
+  // UIDs. See explanation of legacy_uid_t above.
+  typedef uint32_t legacy_uid_t;
+  typedef uint32_t legacy_gid_t;
+
+  struct stat {
+    dev_t st_dev;
+    ino_t st_ino;
+    mode_t st_mode;
+    nlink_t st_nlink;
+    uid_t st_uid;
+    gid_t st_gid;
+    dev_t st_rdev;
+    unsigned long __pad1;
+    off_t st_size;
+    blksize_t st_blksize;
+    int __pad2;
+    blkcnt_t st_blocks;
+    struct timespec st_atim;
+    struct timespec st_mtim;
+    struct timespec st_ctim;
+    int __rr_unused[2];
+  };
+
+  struct stat64 {
```

Comment:
AArch64 doesn't have a stat64 afaict. So why is this here?

Reply:
syscalls.py uses fstatat64 for all architectures. There should probably be a separate entry for fstatat that uses regular struct stat, then this can be dropped. Will try that.

Reply:
Actually, even if we do that, it'll still fail to build because the code generated from syscalls.py is not stratified by architecture. Doing that seems a bit overkill. Should we `typedef sts64 stat` instead on architectures that don't have the stat64 syscall?

Reply:
Actually a typedef doesn't work because `Arch::stat64` is already the syscall, but we could do
```
struct stat64 : public stat {};
```
for 64 bit archs.

Reply:
Ok, yeah, now that I look at it I see x86_64 has the same thing going on. We could clean that up in a followup with the inheritance you suggest but no need to block this.

Reply:
I included the inheritance thing here. Seems to work fine.